public/app/core/utils/richHistory.ts

```diff
@@ -329,6 +329,55 @@ export function notEmptyQuery(query: DataQuery) {
   return false;
 }
 
+export function filterQueriesBySearchFilter(queries: RichHistoryQuery[], searchFilter: string) {
+  const filteredQueries = queries.filter(query => {
+    const searchInComment = query.comment.includes(searchFilter);
```

Comment:
I think it's a good practice to have boolean names start with `is`, `has`, `do`, etc. - e.g. `hasMatchingComment`.

From an algorithmic point of view - if it does have a match, you can return from the function early.